// @kyvejs/types/lcd/kyve/query/v1beta1/account.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v2.6.1
//   protoc               unknown
// source: kyve/query/v1beta1/account.proto
Object.defineProperty(exports, "__esModule", { value: true });
exports.QueryAccountClientImpl = exports.QueryAccountServiceName = exports.RedelegationEntry = exports.QueryAccountRedelegationResponse = exports.QueryAccountRedelegationRequest = exports.Funded = exports.QueryAccountFundedListResponse = exports.QueryAccountFundedListRequest = exports.DelegationUnbonding = exports.QueryAccountDelegationUnbondingsResponse = exports.QueryAccountDelegationUnbondingsRequest = exports.QueryAccountAssetsResponse = exports.QueryAccountAssetsRequest = exports.protobufPackage = void 0;
/* eslint-disable */
var wire_1 = require("@bufbuild/protobuf/wire");
var pagination_1 = require("../../../cosmos/base/query/v1beta1/pagination");
var coin_1 = require("../../../cosmos/base/v1beta1/coin");
var query_1 = require("./query");
exports.protobufPackage = "kyve.query.v1beta1";
function createBaseQueryAccountAssetsRequest() {
    return { address: "" };
}
exports.QueryAccountAssetsRequest = {
    encode: function (message, writer) {
        if (writer === void 0) { writer = new wire_1.BinaryWriter(); }
        if (message.address !== "") {
            writer.uint32(10).string(message.address);
        }
        return writer;
    },
    decode: function (input, length) {
        var reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        var end = length === undefined ? reader.len : reader.pos + length;
        var message = createBaseQueryAccountAssetsRequest();
        while (reader.pos < end) {
            var tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.address = reader.string();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON: function (object) {
        return { address: isSet(object.address) ? globalThis.String(object.address) : "" };
    },
    toJSON: function (message) {
        var obj = {};
        if (message.address !== "") {
            obj.address = message.address;
        }
        return obj;
    },
    create: function (base) {
        return exports.QueryAccountAssetsRequest.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial: function (object) {
        var _a;
        var message = createBaseQueryAccountAssetsRequest();
        message.address = (_a = object.address) !== null && _a !== void 0 ? _a : "";
        return message;
    },
};
function createBaseQueryAccountAssetsResponse() {
    return {
        balance: "0",
        delegation: "0",
        delegation_unbonding: "0",
        delegation_rewards: [],
        commission_rewards: [],
        protocol_funding: [],
    };
}
exports.QueryAccountAssetsResponse = {
    encode: function (message, writer) {
        if (writer === void 0) { writer = new wire_1.BinaryWriter(); }
        if (message.balance !== "0") {
            writer.uint32(8).uint64(message.balance);
        }
        if (message.delegation !== "0") {
            writer.uint32(16).uint64(message.delegation);
        }
        if (message.delegation_unbonding !== "0") {
            writer.uint32(24).uint64(message.delegation_unbonding);
        }
        for (var _i = 0, _a = message.delegation_rewards; _i < _a.length; _i++) {
            var v = _a[_i];
            coin_1.Coin.encode(v, writer.uint32(50).fork()).join();
        }
        for (var _b = 0, _c = message.commission_rewards; _b < _c.length; _b++) {
            var v = _c[_b];
            coin_1.Coin.encode(v, writer.uint32(58).fork()).join();
        }
        for (var _d = 0, _e = message.protocol_funding; _d < _e.length; _d++) {
            var v = _e[_d];
            coin_1.Coin.encode(v, writer.uint32(66).fork()).join();
        }
        return writer;
    },
    decode: function (input, length) {
        var reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        var end = length === undefined ? reader.len : reader.pos + length;
        var message = createBaseQueryAccountAssetsResponse();
        while (reader.pos < end) {
            var tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 8) {
                        break;
                    }
                    message.balance = reader.uint64().toString();
                    continue;
                }
                case 2: {
                    if (tag !== 16) {
                        break;
                    }
                    message.delegation = reader.uint64().toString();
                    continue;
                }
                case 3: {
                    if (tag !== 24) {
                        break;
                    }
                    message.delegation_unbonding = reader.uint64().toString();
                    continue;
                }
                case 6: {
                    if (tag !== 50) {
                        break;
                    }
                    message.delegation_rewards.push(coin_1.Coin.decode(reader, reader.uint32()));
                    continue;
                }
                case 7: {
                    if (tag !== 58) {
                        break;
                    }
                    message.commission_rewards.push(coin_1.Coin.decode(reader, reader.uint32()));
                    continue;
                }
                case 8: {
                    if (tag !== 66) {
                        break;
                    }
                    message.protocol_funding.push(coin_1.Coin.decode(reader, reader.uint32()));
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON: function (object) {
        return {
            balance: isSet(object.balance) ? globalThis.String(object.balance) : "0",
            delegation: isSet(object.delegation) ? globalThis.String(object.delegation) : "0",
            delegation_unbonding: isSet(object.delegation_unbonding) ? globalThis.String(object.delegation_unbonding) : "0",
            delegation_rewards: globalThis.Array.isArray(object === null || object === void 0 ? void 0 : object.delegation_rewards)
                ? object.delegation_rewards.map(function (e) { return coin_1.Coin.fromJSON(e); })
                : [],
            commission_rewards: globalThis.Array.isArray(object === null || object === void 0 ? void 0 : object.commission_rewards)
                ? object.commission_rewards.map(function (e) { return coin_1.Coin.fromJSON(e); })
                : [],
            protocol_funding: globalThis.Array.isArray(object === null || object === void 0 ? void 0 : object.protocol_funding)
                ? object.protocol_funding.map(function (e) { return coin_1.Coin.fromJSON(e); })
                : [],
        };
    },
    toJSON: function (message) {
        var _a, _b, _c;
        var obj = {};
        if (message.balance !== "0") {
            obj.balance = message.balance;
        }
        if (message.delegation !== "0") {
            obj.delegation = message.delegation;
        }
        if (message.delegation_unbonding !== "0") {
            obj.delegation_unbonding = message.delegation_unbonding;
        }
        if ((_a = message.delegation_rewards) === null || _a === void 0 ? void 0 : _a.length) {
            obj.delegation_rewards = message.delegation_rewards.map(function (e) { return coin_1.Coin.toJSON(e); });
        }
        if ((_b = message.commission_rewards) === null || _b === void 0 ? void 0 : _b.length) {
            obj.commission_rewards = message.commission_rewards.map(function (e) { return coin_1.Coin.toJSON(e); });
        }
        if ((_c = message.protocol_funding) === null || _c === void 0 ? void 0 : _c.length) {
            obj.protocol_funding = message.protocol_funding.map(function (e) { return coin_1.Coin.toJSON(e); });
        }
        return obj;
    },
    create: function (base) {
        return exports.QueryAccountAssetsResponse.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial: function (object) {
        var _a, _b, _c, _d, _e, _f;
        var message = createBaseQueryAccountAssetsResponse();
        message.balance = (_a = object.balance) !== null && _a !== void 0 ? _a : "0";
        message.delegation = (_b = object.delegation) !== null && _b !== void 0 ? _b : "0";
        message.delegation_unbonding = (_c = object.delegation_unbonding) !== null && _c !== void 0 ? _c : "0";
        message.delegation_rewards = ((_d = object.delegation_rewards) === null || _d === void 0 ? void 0 : _d.map(function (e) { return coin_1.Coin.fromPartial(e); })) || [];
        message.commission_rewards = ((_e = object.commission_rewards) === null || _e === void 0 ? void 0 : _e.map(function (e) { return coin_1.Coin.fromPartial(e); })) || [];
        message.protocol_funding = ((_f = object.protocol_funding) === null || _f === void 0 ? void 0 : _f.map(function (e) { return coin_1.Coin.fromPartial(e); })) || [];
        return message;
    },
};
function createBaseQueryAccountDelegationUnbondingsRequest() {
    return { pagination: undefined, address: "" };
}
exports.QueryAccountDelegationUnbondingsRequest = {
    encode: function (message, writer) {
        if (writer === void 0) { writer = new wire_1.BinaryWriter(); }
        if (message.pagination !== undefined) {
            pagination_1.PageRequest.encode(message.pagination, writer.uint32(10).fork()).join();
        }
        if (message.address !== "") {
            writer.uint32(18).string(message.address);
        }
        return writer;
    },
    decode: function (input, length) {
        var reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        var end = length === undefined ? reader.len : reader.pos + length;
        var message = createBaseQueryAccountDelegationUnbondingsRequest();
        while (reader.pos < end) {
            var tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.pagination = pagination_1.PageRequest.decode(reader, reader.uint32());
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.address = reader.string();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON: function (object) {
        return {
            pagination: isSet(object.pagination) ? pagination_1.PageRequest.fromJSON(object.pagination) : undefined,
            address: isSet(object.address) ? globalThis.String(object.address) : "",
        };
    },
    toJSON: function (message) {
        var obj = {};
        if (message.pagination !== undefined) {
            obj.pagination = pagination_1.PageRequest.toJSON(message.pagination);
        }
        if (message.address !== "") {
            obj.address = message.address;
        }
        return obj;
    },
    create: function (base) {
        return exports.QueryAccountDelegationUnbondingsRequest.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial: function (object) {
        var _a;
        var message = createBaseQueryAccountDelegationUnbondingsRequest();
        message.pagination = (object.pagination !== undefined && object.pagination !== null)
            ? pagination_1.PageRequest.fromPartial(object.pagination)
            : undefined;
        message.address = (_a = object.address) !== null && _a !== void 0 ? _a : "";
        return message;
    },
};
function createBaseQueryAccountDelegationUnbondingsResponse() {
    return { unbondings: [], pagination: undefined };
}
exports.QueryAccountDelegationUnbondingsResponse = {
    encode: function (message, writer) {
        if (writer === void 0) { writer = new wire_1.BinaryWriter(); }
        for (var _i = 0, _a = message.unbondings; _i < _a.length; _i++) {
            var v = _a[_i];
            exports.DelegationUnbonding.encode(v, writer.uint32(10).fork()).join();
        }
        if (message.pagination !== undefined) {
            pagination_1.PageResponse.encode(message.pagination, writer.uint32(18).fork()).join();
        }
        return writer;
    },
    decode: function (input, length) {
        var reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        var end = length === undefined ? reader.len : reader.pos + length;
        var message = createBaseQueryAccountDelegationUnbondingsResponse();
        while (reader.pos < end) {
            var tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.unbondings.push(exports.DelegationUnbonding.decode(reader, reader.uint32()));
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.pagination = pagination_1.PageResponse.decode(reader, reader.uint32());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON: function (object) {
        return {
            unbondings: globalThis.Array.isArray(object === null || object === void 0 ? void 0 : object.unbondings)
                ? object.unbondings.map(function (e) { return exports.DelegationUnbonding.fromJSON(e); })
                : [],
            pagination: isSet(object.pagination) ? pagination_1.PageResponse.fromJSON(object.pagination) : undefined,
        };
    },
    toJSON: function (message) {
        var _a;
        var obj = {};
        if ((_a = message.unbondings) === null || _a === void 0 ? void 0 : _a.length) {
            obj.unbondings = message.unbondings.map(function (e) { return exports.DelegationUnbonding.toJSON(e); });
        }
        if (message.pagination !== undefined) {
            obj.pagination = pagination_1.PageResponse.toJSON(message.pagination);
        }
        return obj;
    },
    create: function (base) {
        return exports.QueryAccountDelegationUnbondingsResponse.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial: function (object) {
        var _a;
        var message = createBaseQueryAccountDelegationUnbondingsResponse();
        message.unbondings = ((_a = object.unbondings) === null || _a === void 0 ? void 0 : _a.map(function (e) { return exports.DelegationUnbonding.fromPartial(e); })) || [];
        message.pagination = (object.pagination !== undefined && object.pagination !== null)
            ? pagination_1.PageResponse.fromPartial(object.pagination)
            : undefined;
        return message;
    },
};
function createBaseDelegationUnbonding() {
    return { amount: "0", creation_time: "0", staker: undefined };
}
exports.DelegationUnbonding = {
    encode: function (message, writer) {
        if (writer === void 0) { writer = new wire_1.BinaryWriter(); }
        if (message.amount !== "0") {
            writer.uint32(8).uint64(message.amount);
        }
        if (message.creation_time !== "0") {
            writer.uint32(16).uint64(message.creation_time);
        }
        if (message.staker !== undefined) {
            query_1.FullStaker.encode(message.staker, writer.uint32(26).fork()).join();
        }
        return writer;
    },
    decode: function (input, length) {
        var reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        var end = length === undefined ? reader.len : reader.pos + length;
        var message = createBaseDelegationUnbonding();
        while (reader.pos < end) {
            var tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 8) {
                        break;
                    }
                    message.amount = reader.uint64().toString();
                    continue;
                }
                case 2: {
                    if (tag !== 16) {
                        break;
                    }
                    message.creation_time = reader.uint64().toString();
                    continue;
                }
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.staker = query_1.FullStaker.decode(reader, reader.uint32());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON: function (object) {
        return {
            amount: isSet(object.amount) ? globalThis.String(object.amount) : "0",
            creation_time: isSet(object.creation_time) ? globalThis.String(object.creation_time) : "0",
            staker: isSet(object.staker) ? query_1.FullStaker.fromJSON(object.staker) : undefined,
        };
    },
    toJSON: function (message) {
        var obj = {};
        if (message.amount !== "0") {
            obj.amount = message.amount;
        }
        if (message.creation_time !== "0") {
            obj.creation_time = message.creation_time;
        }
        if (message.staker !== undefined) {
            obj.staker = query_1.FullStaker.toJSON(message.staker);
        }
        return obj;
    },
    create: function (base) {
        return exports.DelegationUnbonding.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial: function (object) {
        var _a, _b;
        var message = createBaseDelegationUnbonding();
        message.amount = (_a = object.amount) !== null && _a !== void 0 ? _a : "0";
        message.creation_time = (_b = object.creation_time) !== null && _b !== void 0 ? _b : "0";
        message.staker = (object.staker !== undefined && object.staker !== null)
            ? query_1.FullStaker.fromPartial(object.staker)
            : undefined;
        return message;
    },
};
function createBaseQueryAccountFundedListRequest() {
    return { address: "" };
}
exports.QueryAccountFundedListRequest = {
    encode: function (message, writer) {
        if (writer === void 0) { writer = new wire_1.BinaryWriter(); }
        if (message.address !== "") {
            writer.uint32(10).string(message.address);
        }
        return writer;
    },
    decode: function (input, length) {
        var reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        var end = length === undefined ? reader.len : reader.pos + length;
        var message = createBaseQueryAccountFundedListRequest();
        while (reader.pos < end) {
            var tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.address = reader.string();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON: function (object) {
        return { address: isSet(object.address) ? globalThis.String(object.address) : "" };
    },
    toJSON: function (message) {
        var obj = {};
        if (message.address !== "") {
            obj.address = message.address;
        }
        return obj;
    },
    create: function (base) {
        return exports.QueryAccountFundedListRequest.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial: function (object) {
        var _a;
        var message = createBaseQueryAccountFundedListRequest();
        message.address = (_a = object.address) !== null && _a !== void 0 ? _a : "";
        return message;
    },
};
function createBaseQueryAccountFundedListResponse() {
    return { funded: [] };
}
exports.QueryAccountFundedListResponse = {
    encode: function (message, writer) {
        if (writer === void 0) { writer = new wire_1.BinaryWriter(); }
        for (var _i = 0, _a = message.funded; _i < _a.length; _i++) {
            var v = _a[_i];
            exports.Funded.encode(v, writer.uint32(10).fork()).join();
        }
        return writer;
    },
    decode: function (input, length) {
        var reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        var end = length === undefined ? reader.len : reader.pos + length;
        var message = createBaseQueryAccountFundedListResponse();
        while (reader.pos < end) {
            var tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.funded.push(exports.Funded.decode(reader, reader.uint32()));
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON: function (object) {
        return {
            funded: globalThis.Array.isArray(object === null || object === void 0 ? void 0 : object.funded) ? object.funded.map(function (e) { return exports.Funded.fromJSON(e); }) : [],
        };
    },
    toJSON: function (message) {
        var _a;
        var obj = {};
        if ((_a = message.funded) === null || _a === void 0 ? void 0 : _a.length) {
            obj.funded = message.funded.map(function (e) { return exports.Funded.toJSON(e); });
        }
        return obj;
    },
    create: function (base) {
        return exports.QueryAccountFundedListResponse.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial: function (object) {
        var _a;
        var message = createBaseQueryAccountFundedListResponse();
        message.funded = ((_a = object.funded) === null || _a === void 0 ? void 0 : _a.map(function (e) { return exports.Funded.fromPartial(e); })) || [];
        return message;
    },
};
function createBaseFunded() {
    return { amounts: [], pool: undefined };
}
exports.Funded = {
    encode: function (message, writer) {
        if (writer === void 0) { writer = new wire_1.BinaryWriter(); }
        for (var _i = 0, _a = message.amounts; _i < _a.length; _i++) {
            var v = _a[_i];
            coin_1.Coin.encode(v, writer.uint32(26).fork()).join();
        }
        if (message.pool !== undefined) {
            query_1.BasicPool.encode(message.pool, writer.uint32(18).fork()).join();
        }
        return writer;
    },
    decode: function (input, length) {
        var reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        var end = length === undefined ? reader.len : reader.pos + length;
        var message = createBaseFunded();
        while (reader.pos < end) {
            var tag = reader.uint32();
            switch (tag >>> 3) {
                case 3: {
                    if (tag !== 26) {
                        break;
                    }
                    message.amounts.push(coin_1.Coin.decode(reader, reader.uint32()));
                    continue;
                }
                case 2: {
                    if (tag !== 18) {
                        break;
                    }
                    message.pool = query_1.BasicPool.decode(reader, reader.uint32());
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON: function (object) {
        return {
            amounts: globalThis.Array.isArray(object === null || object === void 0 ? void 0 : object.amounts) ? object.amounts.map(function (e) { return coin_1.Coin.fromJSON(e); }) : [],
            pool: isSet(object.pool) ? query_1.BasicPool.fromJSON(object.pool) : undefined,
        };
    },
    toJSON: function (message) {
        var _a;
        var obj = {};
        if ((_a = message.amounts) === null || _a === void 0 ? void 0 : _a.length) {
            obj.amounts = message.amounts.map(function (e) { return coin_1.Coin.toJSON(e); });
        }
        if (message.pool !== undefined) {
            obj.pool = query_1.BasicPool.toJSON(message.pool);
        }
        return obj;
    },
    create: function (base) {
        return exports.Funded.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial: function (object) {
        var _a;
        var message = createBaseFunded();
        message.amounts = ((_a = object.amounts) === null || _a === void 0 ? void 0 : _a.map(function (e) { return coin_1.Coin.fromPartial(e); })) || [];
        message.pool = (object.pool !== undefined && object.pool !== null) ? query_1.BasicPool.fromPartial(object.pool) : undefined;
        return message;
    },
};
function createBaseQueryAccountRedelegationRequest() {
    return { address: "" };
}
exports.QueryAccountRedelegationRequest = {
    encode: function (message, writer) {
        if (writer === void 0) { writer = new wire_1.BinaryWriter(); }
        if (message.address !== "") {
            writer.uint32(10).string(message.address);
        }
        return writer;
    },
    decode: function (input, length) {
        var reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        var end = length === undefined ? reader.len : reader.pos + length;
        var message = createBaseQueryAccountRedelegationRequest();
        while (reader.pos < end) {
            var tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.address = reader.string();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON: function (object) {
        return { address: isSet(object.address) ? globalThis.String(object.address) : "" };
    },
    toJSON: function (message) {
        var obj = {};
        if (message.address !== "") {
            obj.address = message.address;
        }
        return obj;
    },
    create: function (base) {
        return exports.QueryAccountRedelegationRequest.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial: function (object) {
        var _a;
        var message = createBaseQueryAccountRedelegationRequest();
        message.address = (_a = object.address) !== null && _a !== void 0 ? _a : "";
        return message;
    },
};
function createBaseQueryAccountRedelegationResponse() {
    return { redelegation_cooldown_entries: [], available_slots: "0" };
}
exports.QueryAccountRedelegationResponse = {
    encode: function (message, writer) {
        if (writer === void 0) { writer = new wire_1.BinaryWriter(); }
        for (var _i = 0, _a = message.redelegation_cooldown_entries; _i < _a.length; _i++) {
            var v = _a[_i];
            exports.RedelegationEntry.encode(v, writer.uint32(10).fork()).join();
        }
        if (message.available_slots !== "0") {
            writer.uint32(16).uint64(message.available_slots);
        }
        return writer;
    },
    decode: function (input, length) {
        var reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        var end = length === undefined ? reader.len : reader.pos + length;
        var message = createBaseQueryAccountRedelegationResponse();
        while (reader.pos < end) {
            var tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.redelegation_cooldown_entries.push(exports.RedelegationEntry.decode(reader, reader.uint32()));
                    continue;
                }
                case 2: {
                    if (tag !== 16) {
                        break;
                    }
                    message.available_slots = reader.uint64().toString();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON: function (object) {
        return {
            redelegation_cooldown_entries: globalThis.Array.isArray(object === null || object === void 0 ? void 0 : object.redelegation_cooldown_entries)
                ? object.redelegation_cooldown_entries.map(function (e) { return exports.RedelegationEntry.fromJSON(e); })
                : [],
            available_slots: isSet(object.available_slots) ? globalThis.String(object.available_slots) : "0",
        };
    },
    toJSON: function (message) {
        var _a;
        var obj = {};
        if ((_a = message.redelegation_cooldown_entries) === null || _a === void 0 ? void 0 : _a.length) {
            obj.redelegation_cooldown_entries = message.redelegation_cooldown_entries.map(function (e) { return exports.RedelegationEntry.toJSON(e); });
        }
        if (message.available_slots !== "0") {
            obj.available_slots = message.available_slots;
        }
        return obj;
    },
    create: function (base) {
        return exports.QueryAccountRedelegationResponse.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial: function (object) {
        var _a, _b;
        var message = createBaseQueryAccountRedelegationResponse();
        message.redelegation_cooldown_entries =
            ((_a = object.redelegation_cooldown_entries) === null || _a === void 0 ? void 0 : _a.map(function (e) { return exports.RedelegationEntry.fromPartial(e); })) || [];
        message.available_slots = (_b = object.available_slots) !== null && _b !== void 0 ? _b : "0";
        return message;
    },
};
function createBaseRedelegationEntry() {
    return { creation_date: "0", finish_date: "0" };
}
exports.RedelegationEntry = {
    encode: function (message, writer) {
        if (writer === void 0) { writer = new wire_1.BinaryWriter(); }
        if (message.creation_date !== "0") {
            writer.uint32(8).uint64(message.creation_date);
        }
        if (message.finish_date !== "0") {
            writer.uint32(16).uint64(message.finish_date);
        }
        return writer;
    },
    decode: function (input, length) {
        var reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        var end = length === undefined ? reader.len : reader.pos + length;
        var message = createBaseRedelegationEntry();
        while (reader.pos < end) {
            var tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 8) {
                        break;
                    }
                    message.creation_date = reader.uint64().toString();
                    continue;
                }
                case 2: {
                    if (tag !== 16) {
                        break;
                    }
                    message.finish_date = reader.uint64().toString();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON: function (object) {
        return {
            creation_date: isSet(object.creation_date) ? globalThis.String(object.creation_date) : "0",
            finish_date: isSet(object.finish_date) ? globalThis.String(object.finish_date) : "0",
        };
    },
    toJSON: function (message) {
        var obj = {};
        if (message.creation_date !== "0") {
            obj.creation_date = message.creation_date;
        }
        if (message.finish_date !== "0") {
            obj.finish_date = message.finish_date;
        }
        return obj;
    },
    create: function (base) {
        return exports.RedelegationEntry.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial: function (object) {
        var _a, _b;
        var message = createBaseRedelegationEntry();
        message.creation_date = (_a = object.creation_date) !== null && _a !== void 0 ? _a : "0";
        message.finish_date = (_b = object.finish_date) !== null && _b !== void 0 ? _b : "0";
        return message;
    },
};
exports.QueryAccountServiceName = "kyve.query.v1beta1.QueryAccount";
var QueryAccountClientImpl = /** @class */ (function () {
    function QueryAccountClientImpl(rpc, opts) {
        this.service = (opts === null || opts === void 0 ? void 0 : opts.service) || exports.QueryAccountServiceName;
        this.rpc = rpc;
        this.AccountAssets = this.AccountAssets.bind(this);
        this.AccountFundedList = this.AccountFundedList.bind(this);
    }
    QueryAccountClientImpl.prototype.AccountAssets = function (request) {
        var data = exports.QueryAccountAssetsRequest.encode(request).finish();
        var promise = this.rpc.request(this.service, "AccountAssets", data);
        return promise.then(function (data) { return exports.QueryAccountAssetsResponse.decode(new wire_1.BinaryReader(data)); });
    };
    QueryAccountClientImpl.prototype.AccountFundedList = function (request) {
        var data = exports.QueryAccountFundedListRequest.encode(request).finish();
        var promise = this.rpc.request(this.service, "AccountFundedList", data);
        return promise.then(function (data) { return exports.QueryAccountFundedListResponse.decode(new wire_1.BinaryReader(data)); });
    };
    return QueryAccountClientImpl;
}());
exports.QueryAccountClientImpl = QueryAccountClientImpl;
function isSet(value) {
    return value !== null && value !== undefined;
}
