// @kyvejs/types/client/kyve/multi_coin_rewards/v1beta1/params.js
"use strict";
// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
// versions:
//   protoc-gen-ts_proto  v2.6.1
//   protoc               unknown
// source: kyve/multi_coin_rewards/v1beta1/params.proto
Object.defineProperty(exports, "__esModule", { value: true });
exports.Params = exports.protobufPackage = void 0;
/* eslint-disable */
var wire_1 = require("@bufbuild/protobuf/wire");
exports.protobufPackage = "kyve.multi_coin_rewards.v1beta1";
function createBaseParams() {
    return { multi_coin_distribution_policy_admin_address: "", multi_coin_distribution_pending_time: "0" };
}
exports.Params = {
    encode: function (message, writer) {
        if (writer === void 0) { writer = new wire_1.BinaryWriter(); }
        if (message.multi_coin_distribution_policy_admin_address !== "") {
            writer.uint32(10).string(message.multi_coin_distribution_policy_admin_address);
        }
        if (message.multi_coin_distribution_pending_time !== "0") {
            writer.uint32(16).uint64(message.multi_coin_distribution_pending_time);
        }
        return writer;
    },
    decode: function (input, length) {
        var reader = input instanceof wire_1.BinaryReader ? input : new wire_1.BinaryReader(input);
        var end = length === undefined ? reader.len : reader.pos + length;
        var message = createBaseParams();
        while (reader.pos < end) {
            var tag = reader.uint32();
            switch (tag >>> 3) {
                case 1: {
                    if (tag !== 10) {
                        break;
                    }
                    message.multi_coin_distribution_policy_admin_address = reader.string();
                    continue;
                }
                case 2: {
                    if (tag !== 16) {
                        break;
                    }
                    message.multi_coin_distribution_pending_time = reader.uint64().toString();
                    continue;
                }
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skip(tag & 7);
        }
        return message;
    },
    fromJSON: function (object) {
        return {
            multi_coin_distribution_policy_admin_address: isSet(object.multi_coin_distribution_policy_admin_address)
                ? globalThis.String(object.multi_coin_distribution_policy_admin_address)
                : "",
            multi_coin_distribution_pending_time: isSet(object.multi_coin_distribution_pending_time)
                ? globalThis.String(object.multi_coin_distribution_pending_time)
                : "0",
        };
    },
    toJSON: function (message) {
        var obj = {};
        if (message.multi_coin_distribution_policy_admin_address !== "") {
            obj.multi_coin_distribution_policy_admin_address = message.multi_coin_distribution_policy_admin_address;
        }
        if (message.multi_coin_distribution_pending_time !== "0") {
            obj.multi_coin_distribution_pending_time = message.multi_coin_distribution_pending_time;
        }
        return obj;
    },
    create: function (base) {
        return exports.Params.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial: function (object) {
        var _a, _b;
        var message = createBaseParams();
        message.multi_coin_distribution_policy_admin_address = (_a = object.multi_coin_distribution_policy_admin_address) !== null && _a !== void 0 ? _a : "";
        message.multi_coin_distribution_pending_time = (_b = object.multi_coin_distribution_pending_time) !== null && _b !== void 0 ? _b : "0";
        return message;
    },
};
function isSet(value) {
    return value !== null && value !== undefined;
}
